consumer/src/mode/resolver/types.rs

```diff
@@ -67,15 +68,38 @@ impl ResolverMessageType {
         account: &mut Account,
     ) -> Result<(), ConsumerError> {
         let ens = Ens::get_ens(Address::from_str(&account.id)?, resolver_consumer_context).await?;
-        if let Some(name) = ens.name.clone() {
+        if let Some(_name) = ens.name.clone() {
             info!("ENS for account: {:?}", ens);
-            let mut account =
-                Account::find_by_id(account.id.clone(), &resolver_consumer_context.pg_pool)
-                    .await?
-                    .ok_or(ConsumerError::AccountNotFound)?;
-            account.label = name;
-            account.image = ens.image.clone();
-            account.upsert(&resolver_consumer_context.pg_pool).await?;
+            // We need to update the account metadata
+            self.update_account_metadata(
+                resolver_consumer_context,
+                account.id.clone(),
+                ens.clone(),
+            )
+            .await?;
+            // We also need to update the atom
+            if let Some(atom_id) = account.atom_id.clone() {
+                self.update_atom_metadata(resolver_consumer_context, &atom_id, ens)
+                    .await?;
+            } else {
+                // We deal with the case where the account atom_id was not set
+                // when the account was created. In this case, we need to query the DB
+                // to find the atom_id, as this update happens in another consumer
+                info!(
+                    "No atom found for account: {:?}, querying the DB...",
+                    account
+                );
+                let account =
+                    Account::find_by_id(account.id.clone(), &resolver_consumer_context.pg_pool)
```

Comment:
I created a ticket to simplify the payload that is passed to the resolver consumer, this is going to simplify this logic